### PR TITLE
17 storybook instead of vitepress

### LINKED ISSUE #17: Stroybook instead of vitepress
https://storybook.js.org/tutorials/intro-to-storybook/vue/en/get-started/

### PR DESCRIPTION
**Pull request by FabianWilms**
**Description**

Replaced hacky vitepress-docs with ✨ storybook ❤️ 

**Reference**

closes #17

### CODE REVIEW

File: src/components/Banner/MucBanner.vue [hunk 1]
```diff
@@ -6,12 +6,22 @@ import { computed } from "vue";
 type bannerType = "info" | "warning" | "emergency";
```
Comment (langehm) [inline src/components/Banner/MucBanner.vue:6]: I overlooked the patternlab-page and found that a [fourth type](https://patternlab.muenchen.space/?p=elements-bb-banner) of banner exists. Maybe we could add this type too in this overhaul.

File: src/components/Banner/MucBanner.vue [hunk 2]
```diff
@@ -6,12 +6,22 @@ import { computed } from "vue";
...
 
 const props = defineProps({
+  /**
+   * Changes the style of the banner. Available types are `info`, `warning` and `emergency`.
+   */
   type: {
     type: String as PropType<bannerType>,
     default: "info",
   },
 });
```
Comment (langehm) [inline src/components/Banner/MucBanner.vue:16]:
```suggestion
const props = withDefaults(
  defineProps<{
    /**
     * Changes the style of the banner. Available types are `info`, `warning` and `emergency`.
     */
    type?: bannerType;
  }>(),
  {
    type: "info",
  }
);
```
I didn't change this in the prettier introduction, but we should use the simpler `pure types` variant when defining props and slots (beneath this, `defineSlots`already uses this variant like any other component).

File: src/components/Callout/MucCallout.vue [hunk 3]
```diff
@@ -40,13 +42,33 @@ type calloutType = "info" | "warning" | "success" | "error";
...
+  /**
+   * Icon shown above the callout. Defaults to Icons matching the type.
+   */
+  icon(): any;
+  /**
+   * Heading of the callout.
+   */
+  header(): any;
+  /**
+   * Content beneath the Heading shown as text.
```
Comment (langehm) [inline src/components/Callout/MucCallout.vue:67]:
```suggestion
   * Icon shown above the callout. Defaults to icons matching the type.
   */
  icon(): any;
  /**
   * Heading of the callout.
   */
  header(): any;
  /**
   * Content beneath the heading shown as text.
```
Two words in lowercase: icons and heading

File: src/components/Comment/MucComment.stories.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        date: "01.01.1970",
+        headline: "Lorem ipsum",
+        text: "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonum eirmod tempor invidunt ut",
+    },
+};
+
+export const Slider = {
+    args: {
+        ...Default.args,
+        variant: "slider"
```
Comment (langehm) [inline src/components/Comment/MucComment.stories.ts:37]:
```suggestion
        rating: 2.3,
        variant: "slider"
```
Just added some variance in the example 😃

File: src/components/Intro/MucIntro.vue [hunk 5]
```diff
@@ -1,9 +1,19 @@
 <script setup lang="ts">
 defineProps({
+  /**
+   * Title of the Intro
+   */
   title: {
     type: String,
   },
 });
```
Comment (langehm) [inline src/components/Intro/MucIntro.vue:9]:
```suggestion
defineProps<{
  /**
   * Title of the Intro
   */
  title: string;
}>();
```
Same as above, using the `pure type` variant.

**Review by langehm — CHANGES_REQUESTED** [review]
Very happy with the introduction of the storybook. Looks a lot more refined and professional. Furthermore, this simplifies the introduction of more components. I added a few minor changes - the storybooks just need some prettier-formating.

File: package.json [hunk 6]
```diff
@@ -30,16 +30,15 @@
...
   "scripts": {
-    "docs:dev": "vitepress dev docs",
-    "docs:build": "vitepress build docs",
-    "docs:serve": "vitepress serve docs",
     "build": "rimraf dist && npm run type-check && npm run build-only",
     "test:unit": "vitest",
     "build-only": "vite build",
     "type-check": "vue-tsc --build --force",
     "lint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --fix --ignore-path .gitignore",
     "format": "prettier --write src/",
```
[the remark this answers is not in the document]
Comment (langehm) [reply package.json:38]: I am currently checking if this includes all the folders (currently added just the newly added files / folders). Will come back on that and push the formated files

[the remark this answers is not in the document]
Comment (langehm) [reply package.json:38]: Files like the `.releaserc.js`, `vite.config.ts`, `vitest.config.ts` could benefit from this too. Maybe a better way would be to introduce a prettier-ignore file and include all files in this project by default.

File: src/components/Banner/MucBanner.vue (hunk 1 above)
Comment (FabianWilms) [reply src/components/Banner/MucBanner.vue:6]: I thought about that, but the specific component is only used once on the whole page and wouldn't have much of an impact in custom webcomponents.

Comment (langehm) [reply src/components/Banner/MucBanner.vue:6]: Fair enough - thought about the same. Just wanted to bring this up.

File: src/components/Banner/MucBanner.stories.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    tags: ['autodocs'],
+    //👇 Our exports that end in "Data" are not stories.
+    excludeStories: /.*Data$/,
+    parameters: {
+        docs: {
+            description: {
+                component:
+`A banner-component used for single-line informations. Available in multiple types.
+
+[🔗 Patternlab-Docs](https://patternlab.muenchen.space/?p=viewall-elements-bannerhttps://patternlab.muenchen.space/?p=viewall-elements-banner)
```
[the remark this answers is not in the document]
Comment (langehm) [reply src/components/Banner/MucBanner.stories.ts]: Resolved in [fda3988](https://github.com/it-at-m/muc-patternlab-vue/pull/48/commits/fda39880390745faf1041717c1f36c5430995daf)

**Review by langehm — APPROVED** [review]
All done.

**Pull request merged**